apps/search/js/search.js

```diff
@@ -60,6 +61,20 @@
         });
       };
 
+      // Initialize the port for connecting to the system app to report metrics.
+      navigator.mozApps.getSelf().onsuccess = function() {
+        var app = this.result;
+        app.connect('app-metrics').then(function onConnAccepted(ports) {
+          ports.forEach(function(port) {
+            self._portMetrics = port;
```

Comment:
Good question. This is also my first time working with ports and it doesn't seem to be that well documented.  A lot of the same code copy/pasted.  What I found out through testing is that ports is an empty array if there is no entry in manifest.webapp in any of the apps for app-metrics.  It also prepends an 'iac-' to everything, which threw me off as well.  We have one entry in the system manifest.webapp.  I changed this to take your suggestion.  I believe your suggestion is safe in our case since the manifest.webapp is fixed by us.  I'm not going to modify the self._port algorithm above (which I pretty much copy/pasted) as I don't know enough about how they are using it and if it makes sense for them to index this way.